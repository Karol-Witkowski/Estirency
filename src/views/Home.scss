.home {
  display: flex;
  box-sizing: border-box;
}