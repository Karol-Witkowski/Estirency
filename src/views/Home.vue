<template>
  <div>
    <SelectMenu />
    <Rate data-cy="rate" />
    <ChartComponent />
  </div>
</template>

<script>
import ChartComponent from '@/components/ChartComponent.vue';
import SelectMenu from '@/components/SelectMenu.vue';
import Rate from '@/components/Rate.vue';

export default {
  name: 'Home',
  components: {
    ChartComponent,
    SelectMenu,
    Rate,
  },
};
</script>

<style src="@/stylesheets/Home.scss" lang="scss" scoped />
