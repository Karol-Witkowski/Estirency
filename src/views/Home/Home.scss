.home {
  display: flex;
  box-sizing: border-box;
  @media only screen and (max-width: 950px) {
    flex-wrap: wrap;
  }
}