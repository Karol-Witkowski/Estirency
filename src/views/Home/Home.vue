<template>
  <main>
    <SelectMenu class="selectMenu"/>
    <Rate class="rate"/>
    <ChartComponent class="chart"/>
  </main>
</template>

<script>
import ChartComponent from '@/components/chartComponent/ChartComponent.vue';
import SelectMenu from '@/components/selectmenu/SelectMenu.vue';
import Rate from '@/components/rate/Rate.vue';

export default {
  name: 'Home',
  components: {
    ChartComponent,
    SelectMenu,
    Rate,
  },
};
</script>

<style src="./Home.scss" scoped lang="scss"/>
