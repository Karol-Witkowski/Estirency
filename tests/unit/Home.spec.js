import { shallowMount } from '@vue/test-utils';
import Home from '@/views/Home/Home.vue';

describe('Home.vue test', () => {
  const wrapper = shallowMount(Home);

  it('Check the name of home view and that 3 child components are rendered', () => {
    expect(wrapper.name()).to.match(/Home/);
    expect(wrapper.findAll('.SelectMenu').exists());
    expect(wrapper.findAll('.Rate').exists());
    expect(wrapper.findAll('.ChartComponent').exists());
  });
});
