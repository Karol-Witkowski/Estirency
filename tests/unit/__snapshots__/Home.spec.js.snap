// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue test Render correctly 1`] = `
<div>
  <selectmenu-stub></selectmenu-stub>
  <rate-stub data-cy="rate"></rate-stub>
  <chartcomponent-stub></chartcomponent-stub>
</div>
`;
